File: Aula06/Exercícios/Dinheiro/cash.py
from cs50 import get_float


def main():
    # quarter, dime, nickel, penny
    q = 25
    d = 10
    n = 5
    p = 1
    contmoeda = 0
    valor = get_troco()
    troco = valor * 100  # converter reais pra centavos

    # utiliza as moedas das maiores para as menores
    while troco >= q:
        troco -= q
        contmoeda += 1
    while troco >= d:
        troco -= d
        contmoeda += 1
    while troco >= n:
        troco -= n
        contmoeda += 1
    while troco >= p:
        troco -= p
        contmoeda += 1
    print(contmoeda)


def get_troco():
    # garante que o valor passado seja superior a 0 reais
    while True:
        r = get_float("Valor do troco: ")
        if r > 0:
            break
    return r


main()

File: Week01/Exercises/Cash/cash.c
// Criar um programa que dê o troco no menor número de moedas possível
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //q = quarter, d = dime, n = nickel, p = penny
    float real = 0;
    int q = 25, d = 10, n = 5, p = 1;

    //apenas valores positivos
    do
    {
        real = get_float("Valor do troco: ");
    }
    while (real <= 0);

    //converter de reais para centavos
    int cent = round(real * 100);

    //contar as moedas, cont = contador
    int cont = 0;
    while (cent >= q)
    {
        cent = cent - q;
        cont++;
    }
    while (cent >= d)
    {
        cent = cent - d;
        cont++;
    }
    while (cent >= n)
    {
        cent = cent - n;
        cont++;
    }
    while (cent >= p)
    {
        cent = cent - p;
        cont++;
    }

    printf("%i\n", cont);
}